homeassistant/components/sensor/ambient_station.py

```diff
@@ -0,0 +1,205 @@
+        sys_units = config[CONF_UNITS]
+    elif hass.config.units.is_metric:
+        sys_units = UNITS_SI
+    else:
+        sys_units = UNITS_US
+
+    for condition in config[CONF_MONITORED_CONDITIONS]:
+        # create a sensor object for each monitored condition in the yaml file
+        sensor_params = SENSOR_TYPES[condition]
+        name = sensor_params[SENSOR_NAME]
+        units = sensor_params[SENSOR_UNITS]
+        if isinstance(units, list):
+            units = sensor_params[SENSOR_UNITS][UNIT_SYSTEM[sys_units]]
+
+        sensor_list.append(AmbientWeatherSensor(station_data, condition, name,
+                                                units))
+
+    add_devices(sensor_list)
+
+
+class AmbientWeatherSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, station_data, condition, name, units):
+        """Initialize the sensor."""
+        self._state = None
+        self.station_data = station_data
+        self._condition = condition
+        self._name = name
+        self._units = units
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._units
+
+    def update(self):
+        """Fetch new state data for the sensor.
+
+        This is the only method that should fetch new data for Home Assistant.
+        """
+        _LOGGER.debug("Getting data for sensor: %s", self._name)
+        data = self.station_data.get_data()
+        if data is None:
+            # update likely got throttled and returned None, so use the cached
+            # data from the station_data object
+            self._state = self.station_data.data[self._condition]
+        else:
+            if self._condition in data:
+                self._state = data[self._condition]
+            else:
+                _LOGGER.warning("%s sensor data not available from the "
+                                "station", self._condition)
+
+        _LOGGER.debug("Sensor: %s | Data: %s", self._name, self._state)
+
+
+class AmbientStationData:
+    """Class to interface with ambient-api library."""
+
+    def __init__(self, api_key, app_key, interval):
+        """Initialize station data object."""
+        self._api_key = api_key
+        self._app_key = app_key
+
+        self._interval = interval
+
+        self._api_keys = {
+            'AMBIENT_ENDPOINT':
+            'https://api.ambientweather.net/v1',
+            'AMBIENT_API_KEY': self._api_key,
+            'AMBIENT_APPLICATION_KEY': self._app_key,
+            'log_level': 'DEBUG'
+        }
+
+        self.data = None
+        self._station = None
+        self._api = None
+        self._devices = None
+
+        self.get_data = Throttle(self._interval)(self.update)
```

Comment:
I'm not sure I understand this comment.  Are you saying to set the throttle time argument to the module level constant so that the user can't override it from the config?

Reply:
The user can still override it, but the entity platform helper will handle this. This platform uses the home assistant built in polling which is controlled by the entity platform helper.

Reply:
@MartinHjelmare I made all the requested changes and i've committed back to my fork.  

Thank you for the help (I'm a first timer)!  Let me know if there is anything else I need to change.